concent_api/core/models.py

```diff
@@ -485,3 +493,478 @@ def clean(self):
+        WrongKey = 'WrongKey'
+        WrongAddress = 'WrongAddress'
+        WrongEnvironment = 'WrongEnvironment'
+        NoSourceCode = 'NoSourceCode'
+        WrongDockerImages = 'WrongDockerImages'
+        ConcentRequired = 'ConcentRequired'
+        ConcentDisabled = 'ConcentDisabled'
+
+    if not len(CannotComputeTaskReason.__members__) == len(message.tasks.CannotComputeTask.REASON.__members__):
+        raise ValidationError({
+            'CannotComputeTaskReason': 'CannotComputeTaskReason Enum has not same fields as golem_messages.message.tasks.CannotComputeTask.REASON'
+        })
+
+    for model_item, golem_message_item in zip(CannotComputeTaskReason.__members__, message.tasks.CannotComputeTask.REASON.__members__):
+        if not model_item == golem_message_item:
+            raise ValidationError({
+                'CannotComputeTaskReason': 'CannotComputeTaskReason Enum has not same fields as golem_messages.message.tasks.CannotComputeTask.REASON'
+            })
+
+    task_to_compute = ForeignKey(StoredTaskToCompute)
+    reason = CharField(max_length=32, choices=CannotComputeTaskReason.choices())
+
+
+class StoredReportComputedTask(AbstractMessageModel):
+
+    ITEMS = [
+        'result_type',
+        'node_name',
+        'address',
+        'node_info',
+        'port',
+        'key_id',
+        'extra_data',
+        'eth_account',
+        'task_to_compute',
+        'size',
+        'package_hash',
+        'multihash',
+        'secret',
+        'options',
+        'header',
+        'sig',
+    ]
+
+    if not len(ITEMS) == len(message.tasks.ReportComputedTask().__slots__):
+        raise ValidationError({
+            'StoredReportComputedTask': 'StoredReportComputedTask model has not same fields as golem_messages.message.tasks.ReportComputedTask()'
+        })
+
+    for model_item, golem_message_item in zip(ITEMS, message.tasks.ReportComputedTask().__slots__):
+        if not model_item == golem_message_item:
+            raise ValidationError({
+                'StoredReportComputedTask': 'StoredReportComputedTask model has not same fields as golem_messages.message.tasks.ReportComputedTask()'
+            })
+
+    result_type = PositiveSmallIntegerField()
+    node_name = CharField(max_length=32)
+    address = CharField(max_length=15)
+    node_info = JSONField()
+    port = PositiveIntegerField()
+    key_id = BinaryField()
+    extra_data = JSONField()
+    eth_account = CharField(max_length=32)
+    task_to_compute = ForeignKey(StoredTaskToCompute)
+    size = PositiveIntegerField()
+    package_hash = CharField(max_length=HASH_LENGTH)
+    multihash = CharField(max_length=HASH_LENGTH)
+    secret = TextField()
+    options = TextField()
+
+    def clean(self):
+        '''
+        VALIDATION FOR result_type
+        VALIDATION FOR node_name
+        '''
+        try:
+            validate_ipv4_address(self.address)
+        except ValidationError:
+            return ValidationError({
```

Comment:
You should use [GenericIPAddressField](https://docs.djangoproject.com/en/2.0/ref/models/fields/#genericipaddressfield). It automatically validates this.